app/src/androidTest/java/com/example/warriorsocial/ui/settings/SettingsFragmentTest.java

```diff
@@ -0,0 +1,101 @@
+package com.example.warriorsocial.ui.settings;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import static org.junit.Assert.*;
+
+// This test class tests the shared preferences functionality that saves user selected settings
+// from SettingsFragment in internal storage to be accessed again later.
+// In this case, we are only using booleans because each of these methods are linked to a switch
+// that the user either turns on or off.
+// Ready to integrate with main.
+
+@RunWith(AndroidJUnit4.class)
+public class SettingsFragmentTest  {
+
+    public static final String REPLIES_POST = "REPLIES_POST";
+    public static final String REPLIES_COMMENTS = "REPLIES_COMMENTS";
+    public static final String ALL_NOTIFICATIONS = "ALL_NOTIFICATIONS";
+    public static final boolean BOOL_DEFAULT = false;
+
+    private SharedPreferences sharedPreferences;
+
+    @Before
+    public void before() {
+        // Get context of SettingsFragment to make sure we're testing SettingsFragment not test
+        // application context.
+        // This is necessary because we need to run an instrumented unit test for shared preferences.
```

Comment:
Great commenting,
Fully explains the usages of InstrumentedRegistry API.
I used this same API for my unit tests as well